hydromt/gis_utils.py

```diff
@@ -553,3 +558,46 @@ def to_geographic_bbox(bbox, source_crs):
         bbox = Transformer.from_crs(source_crs, target_crs).transform_bounds(*bbox)
 
     return bbox
+
+
+def prepare_pyogrio_reader_filters(
+    fn: IOBase,
+    bbox: GEOM_TYPES | None = None,
+    mask: GEOM_TYPES | None = None,
+    crs: CRS | None = None,
+) -> Tuple[float, float, float, float] | None:
+    """Create a pyogrio-compatible bbox filter.
+
+    Pyogrio does not accept a mask, and requires a bbox in the same CRS as the data.
+    This function takes the possible bbox filter, mask filter and crs of the input data
+    and returns a bbox in the same crs as the data based on the input filters.
+
+    Parameters
+    ----------
+    fn: IOBase,
+        opened file
+    bbox: GeoDataFrame | GeoSeries | BaseGeometry
+        bounding box to filter the data while reading
+    mask: GeoDataFrame | GeoSeries | BaseGeometry
+        mask to filter the data while reading
+    crs: pyproj.CRS
+        coordinate reference system of the bounding box or geometry. If already set,
+        this argument is ignored.
+    """
+    if bbox is not None and mask is not None:
+        raise ValueError(
+            "Both 'bbox' and 'mask' are provided. Please provide only one."
+        )
+    if bbox is None and mask is None:
+        return None
+    source_crs = CRS(read_info(fn).get("crs", "EPSG:4326"))  # assume WGS84
+
+    if mask is not None:
+        bbox = mask
```

Comment:
What's the difference between a mask and a bbox? from this line and the ones above only one is allowed and when mask is the one set we set it to bbox, so I'm not sure what the use of the mask argument is?

Reply:
We want to be able to filter based on both mask and bbox. Unfortunately the current implementation of `pyogrio` only supports a bbox. If only a mask is given, I convert it to a bbox (mask does not have to be rectangular) matching the CRS of the source data so that we do not read some global dataset without supplying some spatial filter first. Then later on in the `read_vector` the mask is used to refine the data. Let me know if this is not clear from the docs.

Reply:
I could also remove mask argument and do the conversion elsewhere

Reply:
Gotcha, this is more of an internal point, so I think it's fine as is, though it might be good to add a short comment for this reasoning, but aside from that, it looks good to me